docs/develop/auth-with-office-dialog-api.md

```diff
@@ -0,0 +1,94 @@
+
+You can use the Dialog APIs to manage this process by using a flow that is similar to the one described for users to sign in. The only differences are:
+
+- If the user hasn't previously granted the application the permissions it needs, she is prompted to do so in the dialog box after signing in.
+- The dialog window sends the access token to the host window either by using `messageParent` to send the stringified access token or by storing the access token where the host window can retrieve it (and using `messageParent` to tell the host window that the token is available). The token has a time limit, but while it lasts, the host window can use it to directly access the user's resources without any further prompting.
+
+Sample add-ins that use the Dialog APIs for this purpose are listed in [Samples](#samples).
+
+## Using authentication libraries with the dialog
+
+The fact that the Office Dialog and the task pane run in different browser, and JavaScript runtime, instances means that you must use many authentication/authorization libraries in the way that is different from how they are used when authentication and authorization can take place in the same window. The following sections describe the main ways that you usually cannot use these libraries and the way that you *can* use them.
+
+### You usually cannot use the library's internal cache to store tokens
+
+Typically, auth-related libraries will provide an in-memory cache where the access token is stored after it is obtained the first time. If subsequent calls to the resource provider (such as Google, Microsoft Graph, Facebook, etc.) are made, the library will first check to see if the token in its cache is expired. If it is unexpired, the library returns the cached token rather than making another round-trip to the STS for a new token.
+
+For example, a library will typically provide both interactive and "silent" methods for getting a token. When you can do both the authentication and the data calls to the resource in the same browser instance, your code calls the "silent" method to obtain a token just before your code adds the token to the data call. The "silent" method checks for an unexpired token in the cache and returns it, if there is one. Otherwise, the "silent" method calls the interactive method which redirects to the STS's login. After login completes, the interactive method returns the token, but also caches it in memory.
+
+But this pattern is not usable in Office add-ins. Since the login occurs in the Office Dialog's browser instance, the token cache is in that instance. But the data calls to the resource, which would call the "silent" method, are in the task pane's browser instance. The library's token cache does not exist in that instance.
```

Comment:
I've rearranged things a bit, but I don't want to remove the paragraph about the silent method. This is something that the reader will recognize and it will help them understand why that part of the library cannot be used.